packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,742 @@
+   */
+  public readonly name: string;
+
+  constructor(name: string) {
+    this.name = name;
+  }
+
+  /**
+   * The worker type name as expected by Job resource.
+   */
+  public toString(): string {
+    return this.name;
+  }
+}
+
+/**
+ * JobCommand specifies the execution environment and the code executed when a job is run.
+ */
+export class JobCommand {
+
+  /**
+   * Create a glueetl JobCommand with the given scriptLocation
+   *
+   * @param scriptLocation specifies the Amazon Simple Storage Service (Amazon S3) path to a script that executes a job.
+   * @param pythonVersion specifies the Python shell version for the ETL job. Versions supported vary depending on GlueVersion.
+   */
+  public static glueEtl(scriptLocation: string, pythonVersion?: PythonVersion) {
+    return new JobCommand(JobCommandName.GLUE_ETL, scriptLocation, pythonVersion);
+  }
+
+  /**
+   * Create a gluestreaming JobCommand with the given scriptLocation
+   *
+   * @param scriptLocation specifies the Amazon Simple Storage Service (Amazon S3) path to a script that executes a job.
+   * @param pythonVersion specifies the Python shell version for the streaming job. Versions supported vary depending on GlueVersion.
+   */
+  public static glueStreaming(scriptLocation: string, pythonVersion?: PythonVersion) {
+    return new JobCommand(JobCommandName.GLUE_STREAMING, scriptLocation, pythonVersion);
+  }
+
+  /**
+   * Create a pythonshell JobCommand with the given scriptLocation and pythonVersion
+   *
+   * @param scriptLocation specifies the Amazon Simple Storage Service (Amazon S3) path to a script that executes a job.
+   * @param pythonVersion the Python version being used to execute a Python shell job.
+   */
+  public static pythonShell(scriptLocation: string, pythonVersion?: PythonVersion) {
+    return new JobCommand(JobCommandName.PYTHON_SHELL, scriptLocation, pythonVersion);
+  }
+
+  /**
+   * The name of the job command e.g. glueetl for an Apache Spark ETL job or pythonshell for a Python shell job.
+   */
+  readonly name: JobCommandName;
+
+  /**
+   * Specifies the Amazon Simple Storage Service (Amazon S3) path to a script that executes a job.
+   */
+  readonly scriptLocation: string;
+
+  /**
+   * The Python version being used to execute a Python shell job.
+   */
+  readonly pythonVersion?: PythonVersion;
+
+  constructor(name: JobCommandName, scriptLocation: string, pythonVersion?: PythonVersion) {
+    this.name = name;
+    this.scriptLocation = scriptLocation;
+    this.pythonVersion = pythonVersion;
+  }
+}
+
+/**
+ * Constants for some of the special parameters used by {@link Job}.
+ *
+ * These constants can be used as argument names in {@link JobProps.defaultArguments}.
+ *
+ * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+ * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+ */
+export enum JobSpecialArgumentNames {
```

Comment:
- See `JobProps.defaultArguments`... these are some of the known argument names that can be used there.
- Making the arguments part of the JobCommand is a big change from the underlying cfn resource which treat the command and the default arguments as separate things, are we sure we want to do that?

Reply:
agreed with Eli, let's move these to typed properties instead of using these flags. job language should be set automatically. there should also probably be some Script class that creates an S3 asset. generated log group and bucket for logging. job bookmark should be an enum